packages/loopring_v3/contracts/aux/bridge/Bridge.sol

```diff
@@ -0,0 +1,605 @@
+    }
+
+    function batchDeposit(
+        address                 from,
+        BridgeTransfer[] memory deposits
+        )
+        external
+        payable
+    {
+        require(from == msg.sender, "UNAUTHORIZED");
+        // Needs to be possible to do all transfers in a single block
+        require(deposits.length <= MAX_NUM_TRANSACTIONS_IN_BLOCK, "MAX_DEPOSITS_EXCEEDED");
+
+        // Transfers to be done
+        InternalBridgeTransfer[] memory transfers = new InternalBridgeTransfer[](deposits.length);
+
+        // Worst case scenario all tokens are different
+        TokenData[] memory tokens = new TokenData[](deposits.length);
+        uint numDistinctTokens = 0;
+
+        // Run over all deposits summing up total amounts per token
+        address token = address(-1);
+        uint tokenIdx = 0;
+        for (uint i = 0; i < deposits.length; i++) {
+            if(token != deposits[i].token) {
+                token = deposits[i].token;
+                tokenIdx = 0;
+                while(tokenIdx < numDistinctTokens && tokens[tokenIdx].token != token) {
+                    tokenIdx++;
+                }
+                if (tokenIdx == numDistinctTokens) {
+                    tokens[tokenIdx].token = token;
+                    tokens[tokenIdx].tokenID = _getTokenID(token);
+                    numDistinctTokens++;
+                }
+            }
+            tokens[tokenIdx].amount = tokens[tokenIdx].amount.add(deposits[i].amount);
+            deposits[i].token = address(tokens[tokenIdx].tokenID);
+
+            transfers[i].owner = deposits[i].owner;
+            transfers[i].tokenID = tokens[tokenIdx].tokenID;
+            transfers[i].amount = deposits[i].amount;
+        }
+
+        // Do a normal deposit per token
+        for(uint i = 0; i < numDistinctTokens; i++) {
+            if (tokens[i].token == address(0)) {
+                require(tokens[i].amount == msg.value, "INVALID_ETH_DEPOSIT");
+            }
+            _deposit(from, tokens[i].token, uint96(tokens[i].amount));
+        }
+
+        // Store the transfers so they can be processed later
+        _storeTransfers(transfers);
+    }
+
+    function beforeBlockSubmission(
+        bytes memory   txsData,
+        bytes calldata callbackData
+        )
+        external
+        onlyFromExchangeOwner
+    {
+        BridgeOperations memory operations = abi.decode(callbackData, (BridgeOperations));
+
+        Context memory ctx = Context({
+            domainSeparator: DOMAIN_SEPARATOR,
+            accountID: accountID,
+            txIdx: 0
+        });
+
+        _processTransfers(ctx, operations.transferOperations, txsData);
+        _processCalls(ctx, operations, txsData);
+
+        // Make sure we have consumed exactly the expected number of transactions
+        require(txsData.length == ctx.txIdx * ExchangeData.TX_DATA_AVAILABILITY_SIZE, "INVALID_NUM_TXS");
```

Comment:
It seems `txIdx` is a counter, not an index.

Reply:
It's used as an index to read all the transactions, but indeed here it is also used as a counter.